apps/rush-lib/src/cli/RushCommandLineParser.ts

```diff
@@ -369,7 +369,7 @@ export class RushCommandLineParser extends CommandLineParser {
     // performs nontrivial work that can throw an exception.  Either the Rush class would need
     // to handle reporting for those exceptions, or else _populateActions() should be moved
     // to a RushCommandLineParser lifecycle stage that can handle it.
-    if (!process.exitCode || process.exitCode > 0) {
+    if (typeof process.exitCode === 'number' && process.exitCode > 0) {
```

Comment:
`!process.exitCode` used to catch the case when `process.exitCode` was `0`. Under that circumstance, we used to call `process.exit(0)`, but now that will throw `process.exit(1)`.

Perhaps this condition should be

```suggestion
    if (process.exitCode !== undefined) {
```

Reply:
@octogonz - you wrote this code initially (I think). Thoughts?

Reply:
First, ever calling `process.exit()` is one of my greatest regrets in this code base heheh. It was a simple dumb decision that will not be quite a lot of work to reverse.  (Sorry everyone!)

I agree with the spirit of this PR that if no exit code was assigned, we shouldn't blindly assume everything is cool. That's the idea here:

https://github.com/microsoft/rushstack/blob/a43b6075450d5e929b73eb823a1502f14105a875/apps/rush-lib/src/cli/RushCommandLineParser.ts#L115-L121

Thus the ideal fix would be like this:
```ts
if (process.exitCode === undefined) {
  throw new InternalError('The operation failed to set process.exitCode');
}
process.exit(process.exitCode);
```

But then we would need to do some testing and fix up any code paths that haven't been doing it right, such as the one in https://github.com/microsoft/rushstack/issues/2203.

Reply:
@mmkal - Would you want to make @octogonz's suggested change in this PR?

Reply:
@iclanton would it be alright to leave it for a separate PR? This change fixes the external user-facing issue of the wrong exit code being used. I'm open to getting to know the code base well enough to make the "ideal" change, but I don't right now. So would prefer to decouple the functional fix from the refactor.

Reply:
Sounds fine to me.

Reply:
I'm good with that